pkgs/by-name/pi/pied/package.nix

```
@@ -0,0 +1,43 @@
+{
+  lib,
+  fetchFromGitHub,
+  flutter327,
+
+  libunwind,
+  gst_all_1,
+  orc,
+}:
+
+flutter327.buildFlutterApplication rec {
+  pname = "pied";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "Elleo";
+    repo = "pied";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-I2p3GIb54r9r/phbKJsz/cFw1ECdwZ2RnCYVxjsHzg0=";
+  };
+
+  pubspecLock = lib.importJSON ./pubspec.lock.json;
+
+  nativeBuildInputs = [
+    libunwind
+    gst_all_1.gstreamermm
+    orc
+  ];
```

```suggestion
  strictDeps = true;

  buildInputs = [
    gst_all_1.gst-plugins-base
  ];
```

Using `strictDeps = true;` will reveal that these dependenies do not belong in `nativeBuildInputs`, since they are not build-time-only dependencies.

Also, even though there are warnings about the `libunwind` .pc file not being found, they are not errors, so `libunwind` doesn't need to be added here.


Looking in the pkg-config errors you might have seen that these were not found:
`gstreamer-1.0.pc`
`gstreamer-app-1.0.pc`



Using `nix-locate` (from the `nix-index` project) to search nixpkgs outputs, we can find that
`gstreamer-1.0.pc` is inside `gst_all_1.gstreamer`
and
`gstreamer-app-1.0.pc` is inside `gst_all_1.gst-plugins-base` 

`gst_all_1.gst-plugins-base` propagates 
`gst_all_1.gstreamer` so only using `gst_all_1.gst-plugins-base` is enough.

Your solution of using `gst_all_1.gstreamermm` only worked because it propagates `gst_all_1.gst-plugins-base`


Why was `orc` added? It doesn't seem to be required.

---

What is the downside of assigning the dependencies to `nativeBuildInputs`?

---

Without setting `strictDeps = true`, there's no difference.
Having `strictDeps = true` requires you to be more exact with the types of dependencies your package has.

In short: `nativeBuildInputs` is for build-time-only stuff, `buildInputs` is for non-build-time deps, e.g. packages with shared libraries

In more detail: (though I'm not an expert)
`nativeBuildInputs` is for build-time-only stuff whose executables should be available on the `PATH`. While `buildInputs` is for stuff that would only ever be executed after the build is finished. Because of this things inside `buildInputs` are not added to the `PATH`.

`strictDeps = true` is not required when just building normally, however, if cross-compiling for another system, the contents of `nativeBuildInputs` will only be executable on the build system, while the contents of `buildInputs` will only be able to be run on the system you're building for.